semantic_inference_ros/app/instance_segmentation_node

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+"""Node that runs openset segmentation."""
+
+import pathlib
+from dataclasses import dataclass, field
+
+import cv2
+
+# additional imports that may not be here
+import numpy as np
+import rclpy
+import spark_config as sc
+import torch
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+
+import semantic_inference.models as models
+import semantic_inference_ros
+from semantic_inference_ros import Conversions, ImageWorkerConfig
+
+
+@dataclass
+class InstanceSegmentationNodeConfig(sc.Config):
+    """Configuration for ClipPublisherNode."""
+
+    worker: ImageWorkerConfig = field(default_factory=ImageWorkerConfig)
+    model: models.InstanceSegmenterConfig = field(
+        default_factory=models.InstanceSegmenterConfig
+    )  # create InstanceSegmenterConfig in models
+    # visualizer: Any = sc.config_field(
+    #     "feature_visualizer", default="component", required=False
+    # )
+
+
+class InstanceSegmentationNode(Node):
+    """Node to run instance segmentation."""
+
+    def __init__(self):
+        """Start subscriber and publisher."""
+        super().__init__("instance_segmentation_node")
+        config_path = (
+            self.declare_parameter("config_path", "").get_parameter_value().string_value
+        )
+        config_path = pathlib.Path(config_path).expanduser().absolute()
+        if not config_path.exists() and config_path != "":
+            self.get_logger().warn(f"config path '{config_path}' does not exist!")
+            self.config = InstanceSegmentationNodeConfig()
+        else:
+            self.get_logger().info(f"Loading config from '{config_path}'")
+            self.config = sc.Config.load(InstanceSegmentationNodeConfig, config_path)
+
+        self.get_logger().info(f"Initializing with {self.config.show()}")
+        device = models.default_device()
+        self._model = models.InstanceSegmenter(self.config.model).to(device)
+        self._model.eval()  # TODO: causing issue with yolo model
+        self.get_logger().info("Finished initializing!")
+
+        self._pub = self.create_publisher(
+            Image, "semantic/image_raw", 1
+        )  # publish segmented image
+        self._worker = semantic_inference_ros.ImageWorker(
+            self, self.config.worker, "color/image_raw", self._spin_once
+        )  # put image in queue for processing
+
+        self._visualizer = "place_holder"
+        # self._visualizer = self.config.visualizer.create()
+        if self._visualizer is not None:
```

Comment:
It's probably better to use a boolean flag in the config to toggle the visualization on and off